### DISCUSSION TITLE
All checkboxes have `checkbox--table-header` class

### DISCUSSION THREAD

**Discussion by xeilen** [tusen-ai/naive-ui#2494] [Q&A] [answered]
Hi,
In DataTable component with type `selection` column, all checkboxes have  `checkbox--table-header` class
https://www.naiveui.com/en-US/light/components/data-table#select

---
![image](https://user-images.githubusercontent.com/58168117/155129619-02e70fbb-1b37-4971-8ea7-b718e1e48fa2.png)

---

So, the question is:
Should this class be in all checkboxes?
In my case, I need an identifier in `thead` for the checkbox. I want to add an event listener for that checkbox.

**Comment by 07akioni** [comment, marked as answer]
> in all checkboxe

<del>No, we will fix it.</del>

I've check the code, it's expected.

![image](https://user-images.githubusercontent.com/18677354/155177459-c1f39176-b212-48a1-89b6-4140af43c2d8.png)

I don't want hover background to affect checkbox.

I may refactor the class name.
